libtiledbsc/src/cli/cli.cc

```diff
@@ -1,14 +1,38 @@
+// TODO fixes build error on VS2019 due to "missing" include in <tiledb/type.h>
+#include <stdexcept>
+
+#include <tiledbsc/soma.h>
+#include <tiledbsc/soma_collection.h>
 #include <tiledbsc/tiledbsc.h>
 
 using namespace tiledbsc;
 
-int main(int argc, char** argv) {
-    (void)argc;
+void walk_soco(std::string_view uri) {
+    auto soco = SOMACollection::open(uri);
+    auto somas = soco.list_somas();
+
+    for (auto& [name, uri] : somas) {
+        printf("soma %s = %s\n", name.c_str(), uri.c_str());
 
-    std::string uri(argv[1]);
-    auto sc = SCGroup(uri);
+        auto soma = SOMA::open(uri);
+        auto arrays = soma.list_arrays();
+        for (auto& [name, uri] : arrays) {
+            printf("  array %s = %s\n", name.c_str(), uri.c_str());
+        }
+    }
+}
+
+int main(int argc, char** argv) {
+    if (argc != 2) {
+        printf("Usage: %s uri\n", argv[0]);
+        return 1;
+    }
 
-    std::cout << "&sc: " << &sc << std::endl;
+    try {
+        walk_soco(argv[1]);
+    } catch (const std::exception& e) {
+        printf("ERROR: %s\n", e.what());
+    }
```

Comment:
a nit, but i do like final newlines ... whenever someone does `cat` of a file, then their shell prompt won't appear after the last LF-less code line.

Reply:
In the future..should we use spdlog here?

Reply:
I noticed the missing LF too. I'll fix that. 

I worked on spdlog today, but the cmake for Windows failed, so saving it for a future PR.